rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -173,16 +175,18 @@ double ROSImageTexture::computeMedianOfBuffer(const std::deque<double> & buffer)
 }
 
 template<typename T>
-std::vector<uint8_t>
-ROSImageTexture::createNewNormalizedBuffer(
+void ROSImageTexture::createNewNormalizedBuffer(
   const T * image_data,
   size_t image_data_size,
   T minValue,
-  T maxValue) const
+  T maxValue,
+  std::shared_ptr<std::vector<uint8_t>> & buffer) const
 {
-  std::vector<uint8_t> buffer;
-  buffer.resize(image_data_size, 0);
-  uint8_t * output_ptr = &buffer[0];
+  if (buffer->size() != image_data_size) {
+    buffer->resize(image_data_size, 0);
+  }
+
+  uint8_t * output_ptr = &((*buffer.get())[0]);
```

Comment:
Is it possible to simplify this to:

```
uint8_t * output_ptr = &buffer[0];
```

Reply:
It's a shared_ptr, not really sure that I can do that

Reply:
Ah, good point.  OK, never mind this comment then.